cajita/unit_test/tstSparseIndexSpace.hpp

```diff
@@ -0,0 +1,441 @@
+            {
+                test[id_find_mirror( i * size * size + j * size + k )] = true;
+            }
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+            {
+
+                EXPECT_EQ( test[i * size * size + j * size + k], true );
+                EXPECT_EQ( id_find_mirror[i * size * size + j * size + k],
+                           id_mirror[i * size * size + j * size + k] );
+            }
+}
+
+void testSparseIndexSpace()
+{
+    constexpr int N = 3;
+    constexpr int size_per_dim = 16 * 4;
+
+    std::array<int, N> size( {size_per_dim, size_per_dim, size_per_dim} );
+    int capacity = size_per_dim * size_per_dim;
+    float rehash_factor = 1.5;
+    SparseIndexSpace<TEST_EXECSPACE> sis( size, capacity, rehash_factor );
+    TileIndexSpace<2, 4, 4 * 4> tis;
+
+    auto cbd = sis.CellBitsPerTileDim;
+    EXPECT_EQ( cbd, 2 );
+
+    auto cnd = sis.CellNumPerTileDim;
+    EXPECT_EQ( cnd, 4 );
+
+    auto cmd = sis.CellMaskPerTileDim;
+    EXPECT_EQ( cmd, 3 );
+
+    auto cbt = sis.CellBitsPerTile;
+    EXPECT_EQ( cbt, 6 );
+
+    auto cnt = sis.CellNumPerTile;
+    EXPECT_EQ( cnt, 64 );
+
+    Kokkos::View<int ***, TEST_DEVICE> qid_res( "query_id", size[0], size[1],
+                                                size[2] );
+    Kokkos::View<int ***, TEST_DEVICE> cid_res( "compute_id", size[0], size[1],
+                                                size[2] );
+
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, size_per_dim ),
+        KOKKOS_LAMBDA( int i ) {
+            for ( int j = 0; j < size_per_dim; j++ )
+                for ( int k = 0; k < size_per_dim; k++ )
+                {
+                    auto tid = sis.insert_cell( i, j, k );
+                    auto qid = sis.query_cell( i, j, k );
+                    auto cid =
+                        tid * sis.CellNumPerTile +
+                        tis.coord_to_offset( ( i & sis.CellMaskPerTileDim ),
+                                             ( j & sis.CellMaskPerTileDim ),
+                                             ( k & sis.CellMaskPerTileDim ) );
+                    qid_res( i, j, k ) = qid;
+                    cid_res( i, j, k ) = cid;
+                }
+        } );
+
+    auto qid_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), qid_res );
+    auto cid_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), cid_res );
+    for ( int i = 0; i < size_per_dim; i++ )
+        for ( int j = 0; j < size_per_dim; j++ )
+            for ( int k = 0; k < size_per_dim; k++ )
+                EXPECT_EQ( qid_mirror( i, j, k ), cid_mirror( i, j, k ) );
+}
+
+//---------------------------------------------------------------------------//
+// RUN TESTS
+//---------------------------------------------------------------------------//
+TEST( sparse_grid, sparse_index_space_test )
+{
+    // testAtomicOr();
```

Comment:
Uncomment these tests so they run every time and make sure we don't hit the bug again.